src/test/ui/lint/issue-67691-unused-field-in-or-pattern.rs

```diff
@@ -0,0 +1,22 @@
+// run-rustfix
+
+#![deny(unused)]
+
+pub enum MyEnum {
+    A { i: i32, j: i32 },
+    B { i: i32, j: i32 },
+}
+
+pub fn foo(x: MyEnum) {
+    use MyEnum::*;
+
+    match x {
+        A { i, j } | B { i, j } => { //~ ERROR unused variable
```

Comment:
I'd like to see an additional case `Some(A { i, j } | B { i, j })` to make sure this works for `#![feature(or_patterns)]` as well. (Applies to the next test also.)

Reply:
Along with the warnings, I get an "error: or-patterns are not fully implemented yet", so should I include that in the expected stderr?

Reply:
I think we can wait with landing this PR until https://github.com/rust-lang/rust/pull/67668 lands and then there shouldn't be such errors.